File: frontend/lib/models/cart_item.dart
class CartItem {
  final int productoId;
  final String nombre;
  int cantidad;
  final double precio;
  final String? opciones;

  CartItem({
    required this.productoId,
    required this.nombre,
    required this.cantidad,
    required this.precio,
    this.opciones,
  });

  double get subtotal => precio * cantidad;

  factory CartItem.fromJson(Map<String, dynamic> json) {
    return CartItem(
      productoId: json['productoId'],
      nombre: json['nombre'],
      cantidad: json['cantidad'],
      precio: json['precio'].toDouble(),
      opciones: json['opciones'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'productoId': productoId,
      'nombre': nombre,
      'cantidad': cantidad,
      'precio': precio,
      'opciones': opciones,
    };
  }

  CartItem copyWith({
    int? productoId,
    String? nombre,
    int? cantidad,
    double? precio,
    String? opciones,
  }) {
    return CartItem(
      productoId: productoId ?? this.productoId,
      nombre: nombre ?? this.nombre,
      cantidad: cantidad ?? this.cantidad,
      precio: precio ?? this.precio,
      opciones: opciones ?? this.opciones,
    );
  }
}

File: backend/src/main/java/com/wabizone/ecommerce/models/CartItem.java
package com.wabizone.ecommerce.models;

public class CartItem {
    
    private Long productoId;
    private String nombre;
    private Integer cantidad;
    private Double precio;
    private String opciones;
    
    public CartItem() {}
    
    public CartItem(Long productoId, String nombre, Integer cantidad, Double precio) {
        this.productoId = productoId;
        this.nombre = nombre;
        this.cantidad = cantidad;
        this.precio = precio;
    }
    
    public CartItem(Long productoId, String nombre, Integer cantidad, Double precio, String opciones) {
        this.productoId = productoId;
        this.nombre = nombre;
        this.cantidad = cantidad;
        this.precio = precio;
        this.opciones = opciones;
    }
    
    public Long getProductoId() {
        return productoId;
    }
    
    public void setProductoId(Long productoId) {
        this.productoId = productoId;
    }
    
    public String getNombre() {
        return nombre;
    }
    
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    
    public Integer getCantidad() {
        return cantidad;
    }
    
    public void setCantidad(Integer cantidad) {
        this.cantidad = cantidad;
    }
    
    public Double getPrecio() {
        return precio;
    }
    
    public void setPrecio(Double precio) {
        this.precio = precio;
    }
    
    public String getOpciones() {
        return opciones;
    }
    
    public void setOpciones(String opciones) {
        this.opciones = opciones;
    }
}